File: src/main/java/net/casualuhc/uhcmod/command/MinesweeperCommand.java
package net.casualuhc.uhcmod.command;

import com.mojang.brigadier.CommandDispatcher;
import net.casualuhc.uhcmod.utils.screen.MinesweeperScreen;
import net.minecraft.server.command.ServerCommandSource;

import static net.minecraft.server.command.CommandManager.literal;

public class MinesweeperCommand {
	public static void register(CommandDispatcher<ServerCommandSource> dispatcher) {
		dispatcher.register(literal("minesweeper").executes(c -> {
			c.getSource().getPlayerOrThrow().openHandledScreen(MinesweeperScreen.createScreenFactory());
			return 1;
		}));
	}
}

File: src/main/kotlin/net/casualuhc/uhc/commands/MinesweeperCommand.kt
package net.casualuhc.uhc.commands

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import net.casualuhc.uhc.screen.MinesweeperScreen
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands

object MinesweeperCommand: Command {
    override fun register(dispatcher: CommandDispatcher<CommandSourceStack>) {
        dispatcher.register(Commands.literal("minesweeper").executes(this::execute))
    }

    private fun execute(context: CommandContext<CommandSourceStack>): Int {
        context.source.playerOrException.openMenu(MinesweeperScreen.createScreenFactory())
        return 1
    }
}